### PR TITLE
Api 7352 update vv open api

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
# Description

Customers noticed some fields in the disability_rating and service_history endpoints were nullable when they did not expect them to be. This story documents the exact specification of these services' return objects in openapi

# Specifications

**Disability Rating Fields**

- data
  - [x] can it be empty? - YES
  - [x] required - YES
- data.id
  - [x] required - YES
  - [x] nullable - NO
- data.type
  - [x] required - YES
  - [x] nullable - NO
- data.attributes.combine_disability_rating
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.combine_effective_date
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.legal_effective_date
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.individual_ratings
  - [x] required - YES
- data.attributes.individual_ratings.decision
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.individual_ratings.effective_date
  - [x] required - YES 
  - [x] nullable - YES
- data.attributes.individual_ratings.rating_percentage
  - [x] required - YES
  - [x] nullable - YES

**Service History Fields**

- data 
  - [x] required - YES
- data.id
  - [x] required - YES
  - [x] nullable - NO
- data.attributes
  - [x] required - YES
- data.attributes.first_name
  - [x] required - YES
  - [x] nullable - NO
- data.attributes.last_name
  - [x] required - YES
  - [x] nullable - NO
- data.attributes.branch_of_service
  - [x] required - YES
  - [x] nullable - NO - functionally it may be an empty string
- data.attributes.start_date
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.end_date
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.pay_grade
  - [x] required - YES
  - [x] nullable - NO
- data.attributes.discharge_status
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.separation_reason
  - [x] required - YES
  - [x] nullable - NO
- data.attributes.deployments
  - [x] can it be empty? - yes
  - [x] required - YES
- data.attributes.deployments.start_date
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.deployments.end_date
  - [x] required - YES
  - [x] nullable - YES
- data.attributes.deployments.location
  - [x] required - YES
  - [x] nullable - YES
  
# Helpful Commands
  
```
export OPEN_API_URL="http://localhost:3000/services/veteran_verification/docs/v0/veteran_verification"
```

OR

```
export OPEN_API_URL="https://sandbox-api.va.gov/services/veteran_verification/docs/v0/veteran_verification"
```

```
curl -s $OPEN_API_URL | jq '. | 
    {
        'disability_rating': .paths."/disability_rating".get.responses."200".content."application/json", 
        'disability_rating_component': .components.schemas.DisabilityRating, 
        'disability_rating_attributes_component': .components.schemas.DisabilityRatingAttributes, 
        'individual_rating_component': .components.schemas.individual_ratings,
        'service_history': .paths."/service_history".get.responses."200".content."application/json", 
        'service_history_episode_component': .components.schemas.ServiceHistoryEpisode,
        'deployment_component': .components.schemas.Deployment
    }'
```

### CODE REVIEW

File: modules/veteran_verification/VETERAN_VERIFICATION.yml [hunk 1]
```diff
@@ -256,42 +264,61 @@ components:
...
           type: string
           example: service-history-episodes
         attributes:
           type: object
+          required:
+            - first_name
+            - last_name
+            - start_date
+            - end_date
+            - branch
```
Comment (esubocz-va) [inline modules/veteran_verification/VETERAN_VERIFICATION.yml:287]: In PR comments, this is named `branch_of_service` but `branch` here. Is that intentional?

File: modules/veteran_verification/VETERAN_VERIFICATION.yml [hunk 2]
```diff
@@ -256,42 +264,61 @@ components:
...
           type: string
+          nullable: false
           description: Service History Episode ID
           example: "12312AASDf"
         type:
           type: string
           example: service-history-episodes
         attributes:
           type: object
+          required:
```
Comment (esubocz-va) [inline modules/veteran_verification/VETERAN_VERIFICATION.yml:282]: I see `data.attributes.deployments` listed as required in PR comments, but don't see it in this list.

**Review by esubocz-va — CHANGES_REQUESTED** [review]

**Review by esubocz-va — APPROVED** [review]
Updates LGTM

File: modules/veteran_verification/VETERAN_VERIFICATION.yml [hunk 3]
```diff
@@ -256,47 +264,68 @@ components:
...
               example: "Air Force"
             pay_grade:
               type: string
+              nullable: false
               description: |
                 Defines the level of compensation for a position, normalized across military branches
 
                 Possible values include the concatenation of Pay Plan Code and Pay Grade Code.
                 Pay Plan Code is represented as a two-character code, and can be one of five values:
+                 - CC (Commissioned Corps)
```
Comment (dbrown-va) [inline modules/veteran_verification/VETERAN_VERIFICATION.yml:328]: Looks like maybe an accidental duplicate here?

**Review by dbrown-va — APPROVED** [review]
While we are reviewing some of the underlying data sources this is accurate based xsd's so approving.  Follow up may occur depending on any nullable vs required enforcement at the VV level.

**Pull request merged**